sys-libs/libcxx/libcxx-9999.ebuild

```diff
@@ -51,11 +52,12 @@ pkg_setup() {
 
 src_prepare() {
 	cp -f "${FILESDIR}/Makefile" lib/ || die
+	use elibc_musl && epatch "${FILESDIR}/${P}-musl-support.patch"
```

Comment:
> here you could use append-cppflags it seems;

I've considered that option, but it doesn't work. Though you can build libc++ itself without problem, after libc++ is installed, you still get errors when trying to use its headers in user code, unless you use `-D_LIBCPP_HAS_MUSL_LIBC` for every user program.

Normally libc++'s cmake system handles this automatically by putting this definition in some meta-header. The problem is we're not using cmake here :(

This patch can be ditched after we switch to cmake (I'll be working on that).